src/views/TabView/TabView.js

```diff
@@ -157,7 +157,10 @@ class TabView extends PureComponent<void, Props, void> {
     let renderHeader;
     let renderFooter;
 
-    if (typeof tabBarComponent !== 'undefined') {
+    const { state } = this.props.navigation;
+    const tabBar = this.props.router.getScreenConfig(this.props.childNavigationProps[state.routes[state.index].routeName], 'tabBar');
```

Comment:
I think this should be `this.props.childNavigationProps[state.routes[state.index].key]`

reference - https://github.com/react-community/react-navigation/blob/master/src/withCachedChildNavigation.js#L51